### ISSUE TITLE
About the results reported in the paper.

### ISSUE THREAD

**Issue by HinataKikuchi** [di-dimitrov/propaganda-techniques-in-memes#3]
# Summary
We couldn't reproduce the results reported in the paper.
Could you share the configuration file (YAML) you used in the experiments?
And let me know the tips for reproducing the results.

## Details
The hyperparameters reported in the paper are as follows.

|Name|Value|
|:---:|:---:|
|Batch size|32|
|Optimizer|Adam W|
|Learning rate|5e-5|
|epsilon|1e-8|
|weight decay|0.01|
|Max sequence length|128|
|Number of epochs|37|
|Early stopping|F1-micro on dev set|

When I run the command mmf_run with this hyperprameters, the following results are obtained.

|Model|F-Micro|F1-Macro|
|:---:|:---:|:---:|
|visual BERT|0.0336|0.0159|
|MMBT|0.0580|0.0185|
|ViLBERT|0.0345|0.0235|

However, the paper reports the following.

|Model|F-Micro|F1-Macro|
|:---:|:---:|:---:|
|visual BERT|45.34|11.87|
|MMBT|44.23|8.31|
|ViLBERT|46.76|8.99|

There is a large gap between my results and the paper’s results.

## environment

|Element|Contents|
|:---:|:---:|
|OS|Ubuntu 18.04.6 LTS (Bionic Beaver)|
|CPU|Intel(R) Core(TM) i9-9980XE CPU @ 3.00GHz|
|GPU|NVIDIA TITAN RTX (NVIDIA Corporation Device 1e02 (rev a1))|
|Nvidia driver|470.129.06|
|CUDA|V10.2.89|
|Pytorch|1.10.2|

Thank you for any help,

**Comment by di-dimitrov** [comment]
Hi there!
Seeing the results, it seems like there is something wrong with dataset. Such low results were never seen during training and they normally occur due to some error in the setup.

Thanks for the detailed description of hyperparameters. I would like to ask for a few more details before going into specifics.
Could you provide me information about:

- What dataset were you using (I assume the one in the paper but want to make sure)?
- Did you use this repository for running the MMF model ? If so, did you make any changes ? If not, what was your mmf "dataset" setup?
-  Can you provide me with the full `mmf_run` command used to trigger the training for all 3 models?

Also, to answer your question about configurations - this repository contains all configurations that were used for training. Some might differ a little since there were changes since the paper but the whole setup is present. You can find them in the following directories: `./projects/propaganda/configs` , `./mmf/configs/datasets/propaganda`, `./mmf/datasets/builders/propaganda`. Generally you can just search the repo for `propaganda` and it will list you all directories that contain propaganda in their path (using github file search `Go to File`). From there you can explore the different configs.

**Comment by HinataKikuchi** [comment]
Sorry for the late reply.

I found some mistakes and fixed them.
But, I still can’t reproduce the results.

The updated results are:
 used the mmf fork of https://github.com/di-dimitrov/mmf_propaganda.git
(model=visual_bert, run_type=train_val_test)
|name|value|
|:---:|:---:|
|val/propaganda/multilabel_macro_f1| 0.0507
|val/propaganda/multilabel_micro_f1| 0.1886
|test/propaganda/multilabel_macro_f1| 0.0909
|test/propaganda/multilabel_micro_f1| 0.2442

> 	•	What dataset were you using (I assume the one in the paper but want to make sure)?

I used the dataset downloaded from [di-dimitrov/propaganda-techniques-in-memes/tree/main/data/datasets/propaganda/defaults](https://github.com/di-dimitrov/propaganda-techniques-in-memes/tree/main/data/datasets/propaganda/defaults). 

> 	•	Did you use this repository for running the MMF model ? If so, did you make any changes ? If not, what was your mmf "dataset" setup?

I installed the forked MMF in this repository.
(I found another fork in[ https://github.com/di-dimitrov/mmf_propaganda.git](https://github.com/di-dimitrov/mmf_propaganda.git). Which one should I use?)

>  	•	Can you provide me with the full mmf_run command used to trigger the training for all 3 models?

I used the  following commands.

1. visual bert
```sh
!mmf_run config=./projects/propaganda/configs/visual_bert/reproduce.yaml datasets=propaganda model=visual_bert env.save_dir=./output/visual_bert run_type=train_val_test

```

2. MMBT

```sh
!mmf_run config=./projects/propaganda/configs/mmbt/reproduce.yaml datasets=propaganda model=mmbt env.save_dir=./output/mmbt run_type=train_val_test

```

3. ViLBERT
```sh
!mmf_run config=./projects/propaganda/configs/vilbert/reproduce.yaml datasets=propaganda model=vilbert env.save_dir=./output/vilbert run_type=train_val_test
```

**Comment by di-dimitrov** [comment]
> I installed the forked MMF in this repository.
(I found another fork in[ https://github.com/di-dimitrov/mmf_propaganda.git](https://github.com/di-dimitrov/mmf_propaganda.git). Which one should I use?)

Feel free to use both but I generally prefer: https://github.com/di-dimitrov/propaganda-techniques-in-memes

Glad to see that there is improvement. It seems that most of the configs and data you are using are fine but from the commands I can see you are selecting  `./projects/propaganda/configs/vilbert/reproduce.yaml` as your configuration which is not traditionally in the repository and I could not find a fork where those are committed. 
To help I would need to actually see what are the values there. I ran the scripts after you opened the issues a while ago and got close to the same results. Maybe you are customizing something and it is affecting the results. If you could fork this repo, commit whatever setup you prefer and send me a link to check it out, I will be able to help you more.

**Comment by HinataKikuchi** [comment]
Thank you for your reply.
Sorry for the complication, the ``reproduce.yaml`` used for the previous results is as follows.

(reproduce.yaml)
```yaml
includes:
- ./defaults.yaml

optimizer:
  type: adam_w
  params:
    lr: 5e-5
    eps: 1e-8
    weight_decay: 0.01

scheduler: 
  type: warmup_linear
  params:
    num_warmup_steps: 0
    num_training_steps: ${training.max_updates}

training:
  batch_size: 32
  lr_scheduler: true
  max_updates: 900
  evaluation_interval: 400
  find_unused_parameters: true
  early_stop:
    criteria: propaganda/multilabel_micro_f1
    minimize: false

checkpoint:
  resume_pretrained: true
  resume_zoo: visual_bert.pretrained.coco
```

**Comment by di-dimitrov** [comment]
I am sorry for the long time it took to respond. The presented configurations seem correct. I have ran the models a month ago directly from the repository without doing any changes and the results were within expected ranges. My suggestion is to separately clone a clean version of the repository and run everything. If the results are normal carefully review changes (even minor ones). 

I will close the issue as a long time has passed. Please feel free to open a new issue if any problem is present.
